File: MATLAB/compare_replicates.m
% read Sosik chlorophyll data from GitHub
urlwrite('https://raw.githubusercontent.com/WHOIGit/nes-lter-examples/master/data/neslter_sosik_chl.csv', 'local.csv');
T = readtable('local.csv');

% limit to cruise EN608 WSW samlples
T = T(strcmp(T.cruise, 'EN608'),:);
T = T(strcmp(T.filter_mesh_size, '>0'),:);

% separate out the a and b replicates
rep_a = T(strcmp(T.replicate, 'a'),:);
rep_b = T(strcmp(T.replicate, 'b'),:);

% merge rep_a and rep_b to put replicates side by side
for ii = 1:size(rep_a,1)
   temp = rep_b((rep_b.cast==rep_a.cast(ii) & rep_b.niskin==rep_a.niskin(ii)),:);
   if ~isempty(temp)
       rep_b_match2a(ii,:) = temp;
   end
end

% plot a against b
plot(rep_a.chl, rep_b_match2a.chl, '.')

File: R/compare_replicates.R
library(tidyverse)

BASE.URL <- 'https://raw.githubusercontent.com/WHOIGit/nes-lter-examples/master/data/'

csv.url <- paste(BASE.URL, 'neslter_sosik_chl.csv', sep='')

# read the data from GitHub
df <- read.csv(csv.url)

# now select just EN608 WSW replicates
df <- df %>%
  filter(cruise == 'EN608') %>%
  filter(filter_mesh_size == '>0')

# split each replicate into its own dataframe
rep.a = df %>% filter(replicate == 'a')
rep.b = df %>% filter(replicate == 'b')

# now convert to wide form for plotting purposes
merged <- merge(rep.a, rep.b, by=c('cruise','cast','niskin'))

chl.x = merged['chl.x']
chl.y = merged['chl.y']

# make a scatter plot of replicate a against replicate b
p <- ggplot(merged, aes(x=chl.x, y=chl.y)) + geom_point() +
  xlab('Replicate A') +
  ylab('Replicate B')

print(p)
